### ISSUE TITLE
dose results not granular

### ISSUE THREAD

**Issue by tiozab** [darwin-eu-dev/DrugUtilisation#455]
the results from addDrugUse() are only overall, not showing the stratification by unit and route, and unit / route / and drug strength pattern. can this options be added as arguments when estimating drug strength?

**Comment by tiozab** [comment]
the granular version by unit, route, and drug strength pattern is available in https://darwin-eu-dev.github.io/DrugUtilisation/reference/dailyDoseCoverage.html yet the reference gives the wrong impression "estimating coverage", please update the name of the function and the description.

**Comment by catalamarti** [comment]
daily dose is stratified by unit and if multiple units appear the naming will be: initial_daily_dose_milligram and initial_daily_dose_milliliter,
route is at conceptSet level, so multiple conceptSets can be provided with the different routes of interest

**Comment by tiozab** [comment]
can you make a function similar to dailydoseCoverage() that works with a cohort input? right now it works with drug_exposure input. (and please keep it because DED relies on this function), so I suggest to make an additional function or amend the addDrugUse with the option to stratify by drug strength pattern (or how would you go about the stratification by drug strength pattern?) I think this is what Dina and @ginberg would need to stratify the dose results into drug strength pattern of their study?

**Comment by ginberg** [comment]
@tiozab yes it would be nice if there was a way to have the dose by pattern with a cohort input. Specific for the pelargonium study we would like to split the dose result by route.

@catalamarti I think it's not possible to add it in this piece of code since, like you said, the route is per concept?
```
conceptName <- "pelargonium"
drugCohortTable %>%
      dplyr::filter(cohort_definition_id == !!getCohortId(drugCohortTable, conceptName)) %>%
      # add drug use variables
      DrugUtilisation::addDrugUse(
        cdm = cdm,
        ingredientConceptId = 42899638, #pelargonium
        conceptSet = drugConceptSet[conceptName],
        duration = TRUE,
        quantity = FALSE,
        dose = TRUE
      ) %>%
      PatientProfiles::summariseResult(
        strata = list("age_group"),
        variables = list(numericVariables = c("duration", "number_exposures", "initial_daily_dose_milligram", "cumulative_dose_milligram")),
        estimates = list(numericVariables = c("min", "q05", "q25", "median", "q75", "q95", "max"))) %>%
      dplyr::collect() %>%
      dplyr::mutate(group_name = conceptName) %>%
      suppressCounts(minCellCount = minCellCount)
```

**Comment by catalamarti** [comment]
@ginberg https://github.com/darwin-eu/CodelistGenerator/issues/179 here @edward-burn worked in a function to stratify a codelist to a certain route, this is part of the development version of CodelistGenerator. When released it will be re-exported in DrugUtilisation #504

**Comment by tiozab** [comment]
@ginberg dailyDoseCoverage() changes name to summariseDoseCoverage() with the next DUS release, that means we need to update as well? https://github.com/darwin-eu-dev/DrugUtilisation/blob/123efa8b23b022eb255ef096e80d7a03bd73a069/R/dailyDose.R#L227

**Comment by tiozab** [comment]
@ginberg I was thinking that you when you run the dailyDoseCoverage() with the cdm$drug_exposure and stratify by route and drug strength pattern you can hand pick the individual patterns and make code lists based upon what could be there (i.e. make the lists according to the pattern you see from amount unit concept id, numerator concept id and denominator concept id and whether values are missing or not) I think this is at the moment the only way to get stratified dose results of a cohort by drug strenght pattern, correct @catalamarti ?

**Comment by catalamarti** [comment]
so @edward-burn was adding some stratify codelist functions to CodelistGenerator, and we were discussing if a pattern specific one was needed. We agreed no, as we assumed that what we want is unit or route specific. But happy to discuss if you think differently @tiozab

**Comment by tiozab** [comment]
I also discussed it with @edward-burn and the reason why it cannot be stratified at the codelistgenerator level is because the patterns are not at the right level for stratification by codelistgenerator.
Yet, given the example of Gers and Dinas study, dose output from cohorts that are stratified by pattern would be needed for the future.
